File: src/main/java/net/reflact/engine/classes/ClassManager.java
package net.reflact.engine.classes;

import net.reflact.engine.spells.Spell;
import java.util.Map;
import java.util.HashMap;
import java.util.List;
import java.util.ArrayList;

public class ClassManager {
    private final Map<ReflactClass, Map<Integer, List<String>>> abilities = new HashMap<>();

    public void init() {
        // Define Abilities for Classes
        // Format: Class -> { Level -> [SpellID, SpellID] }
        
        // Warrior: 
        // Lvl 1: Bash
        // Lvl 10: Charge
        Map<Integer, List<String>> warriorSpells = new HashMap<>();
        warriorSpells.put(1, List.of("bash"));
        warriorSpells.put(10, List.of("charge"));
        abilities.put(ReflactClass.WARRIOR, warriorSpells);
        
        // Mage:
        // Lvl 1: Fireball
        // Lvl 5: Teleport (Blink)
        // Lvl 10: Heal
        Map<Integer, List<String>> mageSpells = new HashMap<>();
        mageSpells.put(1, List.of("fireball"));
        mageSpells.put(5, List.of("blink"));
        mageSpells.put(10, List.of("heal"));
        abilities.put(ReflactClass.MAGE, mageSpells);
        
        // etc...
    }
    
    public List<String> getAvailableSpells(ReflactClass clazz, int level) {
        List<String> spells = new ArrayList<>();
        Map<Integer, List<String>> classSpells = abilities.get(clazz);
        if (classSpells == null) return spells;
        
        for (Map.Entry<Integer, List<String>> entry : classSpells.entrySet()) {
            if (level >= entry.getKey()) {
                spells.addAll(entry.getValue());
            }
        }
        return spells;
    }
}

File: src/main/kotlin/net/reflact/engine/classes/ClassManager.kt
package net.reflact.engine.classes

class ClassManager {
    private val abilities: MutableMap<ReflactClass, Map<Int, List<String>>> = HashMap()

    fun init() {
        // Define Abilities for Classes
        // Format: Class -> { Level -> [SpellID, SpellID] }

        // Warrior:
        // Lvl 1: Bash
        // Lvl 10: Charge
        val warriorSpells = mapOf(
            1 to listOf("bash"),
            10 to listOf("charge")
        )
        abilities[ReflactClass.WARRIOR] = warriorSpells

        // Mage:
        // Lvl 1: Fireball
        // Lvl 5: Teleport (Blink)
        // Lvl 10: Heal
        val mageSpells = mapOf(
            1 to listOf("fireball"),
            5 to listOf("blink"),
            10 to listOf("heal")
        )
        abilities[ReflactClass.MAGE] = mageSpells

        // etc...
    }

    fun getAvailableSpells(clazz: ReflactClass, level: Int): List<String> {
        val spells = ArrayList<String>()
        val classSpells = abilities[clazz] ?: return spells

        for ((lvl, spellList) in classSpells) {
            if (level >= lvl) {
                spells.addAll(spellList)
            }
        }
        return spells
    }
}
